### ISSUE TITLE
Parsing issue: "NOT BETWEEN" does not parse

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#142]
Some of our queries use `NOT BETWEEN` (https://www.dofactory.com/sql/where-between). This results in a parsing error.

Input:
```
SELECT
    business_type
FROM
    benchmark_summaries
WHERE
    avg_click_rate NOT BETWEEN 0 and 1
```

Results:
```
(sqlfluff-3.6.4) datascience-benchmark-summary$ sqlfluff lint test.sql
== [test.sql] FAIL
L:   4 | P:   5 | L003 | Indentation not consistent with line #3
L:   6 | P:   5 | L003 | Indentation not consistent with line #5
L:   6 | P:  20 |  PRS | Found unparsable segment @L006P020: 'NOT BETWEEN 0 and 1\n...'
```

**Comment by alanmcruickshank** [comment]
Great catch. I'm not sure we support `BETWEEN` even yet - but I'll see if we can do `NOT BETWEEN` at the same time.

**Comment by alanmcruickshank** [comment]
This is now closed in #155
